packages/material-ui/src/FilledInput/FilledInput.js

```diff
@@ -124,13 +124,19 @@ export const styles = theme => {
 };
 
 function FilledInput(props) {
-  const { classes, ...other } = props;
+  const { disableUnderline, classes, ...other } = props;
 
   return (
     <InputBase
       classes={{
         ...classes,
-        root: classNames(classes.root, classes.underline, {}),
+        root: classNames(
+          classes.root,
+          {
+            [classes.underline]: !disableUnderline,
+          },
+          {},
+        ),
         underline: null,
```

Comment:
As far as I know classes are merged. I think someone though this would somehow unset this class (which doesn't even exist on `InputBase`). It should be safe to remove this too.

Reply:
This line is important.